lib/install.js

```diff
@@ -272,8 +272,10 @@ Installer.prototype.run = function (cb) {
         [this, this.saveToDependencies])
     }
   }
-  postInstallSteps.push(
-    [this, this.printInstalled])
+  if (npm.config.get('loglevel') !== 'silent') {
```

Comment:
@iarna, those are the flags I was confused about. [I mentioned below](https://github.com/npm/npm/pull/11784#issuecomment-210131905) that `npmlog` would have to support disabling the line prefix to allow outputting parsable data. I also [devised a scheme](https://github.com/npm/npm/pull/11784#issuecomment-210120127) to force true silence without breaking backwards compatibility.